include/swift/AST/DiagnosticConsumer.h

```diff
@@ -111,6 +112,55 @@ class NullDiagnosticConsumer : public DiagnosticConsumer {
                         ArrayRef<DiagnosticArgument> FormatArgs,
                         const DiagnosticInfo &Info) override;
 };
+
+/// \brief DiagnosticConsumer that funnels diagnostics in certain files to
+/// particular sub-consumers.
```

Comment:
Why is the funneling needed? (I know I already commented about a bigger picture comment, which could help.)
Inheritance, or aggregation? Is it a special kind of DiagnosticConsumer that funnels, or is it an aggregation or ordinary DiagnosticConsumers, a multiplexer in front of a bunch of them? Just asking, don't know the best answer.